api/v1alpha1/mariadb_replication_types.go

```diff
@@ -252,20 +243,38 @@ type ReplicationSpec struct {
 	InitContainer InitContainer `json:"initContainer,omitempty"`
 }
 
-// FillWithDefaults fills the current ReplicationSpec object with DefaultReplicationSpec.
-// This enables having minimal ReplicationSpec objects and provides sensible defaults.
-func (r *ReplicationSpec) FillWithDefaults() {
+// SetDefaults fills the current Replication object with DefaultReplicationSpec.
+// This enables having minimal Replication objects and provides sensible defaults.
+func (r *Replication) SetDefaults(mdb *MariaDB, env *environment.OperatorEnv) error {
+	if r.GtidStrictMode == nil {
+		r.GtidStrictMode = ptr.To(true)
+	}
+	if reflect.ValueOf(r.InitContainer).IsZero() {
+		r.InitContainer = InitContainer{
+			Image: env.MariadbOperatorImage,
+		}
+	}
+
+	autoUpdateDataPlane := ptr.Deref(mdb.Spec.UpdateStrategy.AutoUpdateDataPlane, false)
+	if autoUpdateDataPlane {
+		initBumped, err := docker.SetTagOrDigest(env.MariadbOperatorImage, r.InitContainer.Image)
+		if err != nil {
+			return fmt.Errorf("error bumping replication init image: %v", err)
+		}
+		r.InitContainer.Image = initBumped
+	}
+
 	if r.Primary == nil {
 		primary := *DefaultReplicationSpec.Primary
 		r.Primary = &primary
 	} else {
-		r.Primary.FillWithDefaults()
+		r.Primary.SetDefaults()
```

Comment:
Given that primary will be a struct we can directly call `SetDefaults` here and avoid this conditional defaulting logic.